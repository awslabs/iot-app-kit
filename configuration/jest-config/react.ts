export default {
  setupFilesAfterEnv: ['jest-extended/all'],
  maxWorkers: '50%',
  moduleFileExtensions: ['js', 'ts', 'jsx', 'tsx'],
  collectCoverage: true,
  testEnvironment: 'jsdom',
  transform: {
    '.*\\.(tsx?|jsx?)$': [
      '@swc/jest',
      {
        jsc: {
          transform: {
            react: {
              runtime: 'automatic',
            },
          },
        },
      },
    ],
  },
  transformIgnorePatterns: [],
  moduleNameMapper: {
    '\\.(css|scss)$': 'identity-obj-proxy',
  },
};
