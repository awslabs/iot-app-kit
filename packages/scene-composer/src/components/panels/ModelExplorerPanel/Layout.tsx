import React, { FC } from 'react';
import { FormattedMessage, defineMessages } from 'react-intl';

import LogProvider from '../../../logger/react-logger/log-provider';

const strings = defineMessages({
  NoModelSelected: {
    defaultMessage: 'Select a model ref node to explore the subtree',
    description:
      'A message displayed in a panel to indicate to the user they need to select an object in the scene in order for this panel to be populated.',
  },
});

interface ModelExplorerLayoutProps {
  active: boolean;
}

const ModelExplorerLayout: FC<ModelExplorerLayoutProps> = ({ active, children }) => {
  return (
    <LogProvider namespace='ModelExplorer'>
      {active && children}
      {!active && <FormattedMessage tagName={'p'} {...strings.NoModelSelected} />}
    </LogProvider>
  );
};

export default ModelExplorerLayout;
