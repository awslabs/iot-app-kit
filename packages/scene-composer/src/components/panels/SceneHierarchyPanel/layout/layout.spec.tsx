import React from 'react';
import { render } from '@testing-library/react';

import Layout, { Main, ActionBar, Toolbar } from '.';

describe('<SceneHierarchyPanelLayout />', () => {
  it('should render appropriate structure', () => {
    const { container } = render(
      <Layout>
        <Toolbar>
          Search Box
          <ActionBar>Tools</ActionBar>
        </Toolbar>
        <Main>Main</Main>
      </Layout>,
    );

    expect(container).toMatchSnapshot();
  });
});
