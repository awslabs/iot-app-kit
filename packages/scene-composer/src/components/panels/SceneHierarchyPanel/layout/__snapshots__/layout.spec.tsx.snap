// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SceneHierarchyPanelLayout /> should render appropriate structure 1`] = `
<div>
  <div
    class="tm-hierarchy-panel"
  >
    <div
      class="tm-toolbar"
    >
      Search Box
      <div
        class="tm-action-bar"
      >
        Tools
      </div>
    </div>
    <div
      class="tm-main"
    >
      Main
    </div>
  </div>
</div>
`;
